src/dynamic_honey_badger/mod.rs

```diff
@@ -112,19 +116,54 @@ pub enum Message<N: Rand> {
 }
 
 impl<N: Rand> Message<N> {
-    fn start_epoch(&self) -> u64 {
+    fn era(&self) -> u64 {
         match *self {
-            Message::HoneyBadger(epoch, _) => epoch,
-            Message::KeyGen(epoch, _, _) => epoch,
+            Message::HoneyBadger(era, _) => era,
+            Message::KeyGen(era, _, _) => era,
             Message::SignedVote(ref signed_vote) => signed_vote.era(),
         }
     }
+}
+
+/// Dynamic Honey Badger epoch. It consists of an era and an epoch of Honey Badger that started in
+/// that era. For messages originating from `DynamicHoneyBadger` as opposed to `HoneyBadger`, that
+/// HoneyBadger epoch is `None`.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, Serialize, Deserialize)]
+pub struct Epoch(pub(super) u64, pub(super) Option<u64>);
```

Comment:
Need an `Add` impl for this.

Reply:
I don't think so. From the user's perspective, batches will still have `u64` epoch numbers.

Reply:
Or, I guess it would be sufficient to just add a method which returns the epoch as a u64.

Reply:
I guess it would be sufficient to just add a method which returns the epoch as a u64.

Reply:
Somehow I feel like the user-visible `seqnum` should keep the name `epoch`…

Reply:
`+` won't work on `Epoch`. If you need the next epoch of a batch, please see `SenderQueueableOutput::next_epoch`.